File: src/main/java/me/onixdev/compability/manager/CompatibilityManager.java
package me.onixdev.compability.manager;

import me.onixdev.compability.ICompabilityCheck;
import me.onixdev.compability.impl.LeafCompabilityWorldTicking;
import me.onixdev.compability.impl.ViaBackwardsTransactionPing;

import java.util.List;

public class CompatibilityManager {
    private boolean leafTicking;

    public void setLeafTicking(boolean leafTicking) {
        this.leafTicking = leafTicking;
    }

    public boolean isLeafTicking() {
        return leafTicking;
    }

    private List<ICompabilityCheck> checks = List.of(new LeafCompabilityWorldTicking(),new ViaBackwardsTransactionPing());
    public CompatibilityManager() {
        for (ICompabilityCheck check : checks) {
            check.check(this);
        }
    }
}

File: src/main/java/me/onixdev/compability/manager/CompatibilityManager.kt
package me.onixdev.compability.manager

import me.onixdev.compability.ICompabilityCheck
import me.onixdev.compability.impl.LeafCompabilityWorldTicking
import me.onixdev.compability.impl.PlugmanCompability
import me.onixdev.compability.impl.ViaBackwardsTransactionPing

class CompatibilityManager {
    var isLeafTicking: Boolean = false
    var isHasPlugman: Boolean = false

    private val checks: MutableList<ICompabilityCheck> =
        mutableListOf(LeafCompabilityWorldTicking(), ViaBackwardsTransactionPing(), PlugmanCompability())

    init {
        for (check in checks) {
            check.check(this)
        }
    }
}